test/torchaudio_unittest/rnnt/autograd_impl.py

```diff
@@ -0,0 +1,150 @@
+            self,
+            loss: Callable[..., Tensor],
+            inputs: Tuple[torch.Tensor],
+            *,
+            enable_all_grad: bool = True,
+            nondet_tol: float = 0.0,
+    ):
+        inputs_ = []
+        for i in inputs:
+            if torch.is_tensor(i):
+                i = i.to(dtype=self.dtype, device=self.device)
+                if enable_all_grad:
+                    i.requires_grad = True
+            inputs_.append(i)
+        assert gradcheck(loss, inputs_, eps=1e-03, atol=1e-02, nondet_tol=nondet_tol)
+
+    @parameterized.expand([
+        (get_B1_T10_U3_D4_data, ),
+        (get_B1_T10_U3_D4_data, ),
+        (get_numpy_data_B2_T4_U3_D3, ),
+        (get_numpy_data_B1_T2_U3_D5, ),
+    ])
+    def test_RNNTLoss(self, data_func):
+        data_np = data_func()
+        if type(data_np) == tuple:
+            data_np = data_np[0]
+        data = numpy_to_torch(
+            data=data_np, device=self.device, requires_grad=True
+        )
+        inputs = (
+            data["logits"],
+            data["targets"],
+            data["logit_lengths"],
+            data["target_lengths"],
+        )
+        loss = RNNTLoss(blank=data["blank"])
+        
+        self.assert_grad(loss, inputs, enable_all_grad=False, nondet_tol=1e-10)
+
+    @parameterized.expand([
+        (get_B1_T10_U3_D4_data, ),
+        (get_B1_T10_U3_D4_data, ),
+        (get_numpy_data_B2_T4_U3_D3, ),
+        (get_numpy_data_B1_T2_U3_D5, ),
+    ])
+    def test_rnnt_loss(self, data_func):
+        data_np = data_func()
+        if type(data_np) == tuple:
+            data_np = data_np[0]
+        data = numpy_to_torch(
+            data=data_np, device=self.device, requires_grad=True
+        )
+        inputs = (
+            data["logits"],
+            data["targets"],
+            data["logit_lengths"],
+            data["target_lengths"],
+        )
+        loss = RNNTLoss(blank=data["blank"])
+        
+        self.assert_grad(loss, inputs, enable_all_grad=False, nondet_tol=1e-10)
+
+    @parameterized.expand([
+        (get_B1_T10_U3_D4_data, ),
+        (get_B1_T10_U3_D4_data, ),
+        (get_numpy_data_B2_T4_U3_D3, ),
+        (get_numpy_data_B1_T2_U3_D5, ),
+    ])
+    def test_np_transducer_loss(self, data_func):
+        data_np = data_func()
+        if type(data_np) == tuple:
+            data_np = data_np[0]
+        data = numpy_to_torch(
+            data=data_np, device=self.device, requires_grad=True
+        )
+        inputs = (
+            data["logits"],
+            data["logit_lengths"],
+            data["target_lengths"],
+            data["targets"],
+        )
+        loss = NumpyTransducerLoss(blank=data["blank"])
+        
+        self.assert_grad(loss, inputs, enable_all_grad=False, nondet_tol=1e-10)
+
+    @parameterized.expand([
+        (get_B1_T10_U3_D4_data, ),
+        (get_B1_T10_U3_D4_data, ),
+        (get_numpy_data_B2_T4_U3_D3, ),
+        (get_numpy_data_B1_T2_U3_D5, ),
```

Comment:
both the loss and gradients do not match for this set of data. they match for the other 3 cases

Reply:
Some weird boundary effect?? middle 3 cols are the same, but cols 0 and 4 seem generally swapped with the exception of the final row
```
tensor([4.4957], grad_fn=<_RNNTBackward>) 
tensor([[[[-0.1312, -0.3999,  0.1770,  0.1770,  0.1770],
          [-0.1857,  0.1225, -0.1817,  0.1225,  0.1225],
          [-0.3209,  0.0627,  0.0693,  0.1262,  0.0627]],

         [[ 0.0546, -0.2182,  0.0546,  0.0546,  0.0546],
          [ 0.1207,  0.1207, -0.4830,  0.1207,  0.1207],
          [-0.6926,  0.1687,  0.1865,  0.1687,  0.1687]]]])
tensor(5.0957, grad_fn=<SumBackward0>)
tensor([[[[ 0.1770, -0.3999,  0.1770,  0.1770, -0.1312],
          [ 0.1225,  0.1225, -0.1817,  0.1225, -0.1857],
          [ 0.0627,  0.0627,  0.0693,  0.1262, -0.3209]],

         [[ 0.0546, -0.2182,  0.0546,  0.0546,  0.0546],
          [ 0.1207,  0.1207, -0.4830,  0.1207,  0.1207],
          [ 0.3074,  0.1687,  0.1865,  0.1687, -0.8313]]]])
```